core/src/main/java/com/linecorp/armeria/internal/Http2ObjectEncoder.java

```diff
@@ -70,8 +88,58 @@ protected ChannelFuture doWriteHeaders(int id, int streamId, HttpHeaders headers
         }
 
         // Client starts a new stream.
-        return encoder.writeHeaders(
-                ctx, streamId, ArmeriaHttpUtil.toNettyHttp2(headers, server), 0, endStream, ctx.newPromise());
+        return encoder.writeHeaders(ctx, streamId,
+                                    convert(headers, additionalHeaders, additionalTrailers,
+                                            false, endStream, isTrailer, enableServerHeader, enableDateHeader,
+                                            sessionProtocol, authorityHeader, userAgentHeader),
+                                    0, endStream, ctx.newPromise());
+    }
+
+    private Http2Headers convert(HttpHeaders inputHeaders,
+                                 HttpHeaders additionalHeaders, HttpHeaders additionalTrailers,
+                                 boolean server, boolean endStream, boolean isTrailer,
+                                 boolean enableServerHeader, boolean enableDateHeader,
+                                 @Nullable SessionProtocol sessionProtocol,
+                                 @Nullable String authorityHeader,
+                                 @Nullable String userAgentHeader) {
```

Comment:
Ditto for `userAgentHeader` and `authorityHeader`

Reply:
Removed `userAgentHeader` and `authorityHeader`